src/monmove.cpp

```diff
@@ -1381,6 +1527,73 @@ bool monster::attack_at( const tripoint &p )
     return false;
 }
 
+bool monster::pickup_at( const tripoint &p, item_location &target )
+{
+    //If called through normal pathfinding, then clear looting effect
+    if( p == goal ) {
+        remove_effect( effect_looting );
+
+    }
+
+    //Item has moved...
+    if( p != target.position() ) {
+        return false;
+    }
+
+    inv.push_back( *target.get_item() );
+    item *stored_item = &inv.back();
+
+    //Get weight of the stack or item
+    units::mass weight = target->weight();
+    units::mass capacity = this->weight_capacity();
+
+    int amount_taken = 1;
+
+    int charges = target->charges;
+
+    if( charges >= 1 ) {
+        //Adjust volume and weight for units in a stack
+        units::mass weigh_each = weight / charges;
+        amount_taken = charges + ( weigh_each / capacity );
+
+        //Caps the amount taken to avoid taking large stacks.
+        //TODO: better way to cap the amount taken?
```

Comment:
```suggestion
        amount_taken = std::min( amount_taken, 2 );
```
At least this avoids repeating the value and thereby makes it easier to change it later.